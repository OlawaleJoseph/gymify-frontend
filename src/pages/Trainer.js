import React from 'react';

const Trainer = () => (
  <div>
    Trainer Page
  </div>
);

export default Trainer;
