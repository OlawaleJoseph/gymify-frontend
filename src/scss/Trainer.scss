.profile-card {
  max-width: 350px;
  background-color: #fff;
  box-shadow: 0 0 25px rgba(0, 0, 0, 0.1);
  background-position: center;
  overflow: hidden;
  position: relative;
  cursor: pointer;
  border-radius: 10px;

  img {
    transition: all linear 0.25s;
  }

  .profile-name {
    position: absolute;
    left: 30px;
    bottom: 70px;
    font-size: 30px;
    color: #fff;
    text-shadow: 0 0 20px rgba(0, 0, 0, 0.5);
    font-weight: bold;
    transition: all linear 0.25s;
  }

  .profile-username {
    position: absolute;
    bottom: 50px;
    left: 30px;
    color: #fff;
    font-size: 13px;
    transition: all linear 0.25s;
  }

  .profile-speciality {
    position: absolute;
    bottom: 30px;
    left: 30px;
    color: #fff;
    font-size: 14px;
    font-weight: bold;
    text-transform: capitalize;
    transition: all linear 0.25s;
  }

  &:hover img {
    filter: grayscale(100%);
  }

  &:hover .profile-name {
    bottom: 80px;
  }

  &:hover .profile-username {
    bottom: 60px;
  }

  &:hover .profile-speciality {
    bottom: 40px;
  }
}
