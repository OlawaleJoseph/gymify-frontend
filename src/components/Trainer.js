import React from 'react';
import PropTypes from 'prop-types';
import '../scss/Trainer.scss';
import { Link } from 'react-router-dom';

const Trainer = ({
  id, imgUrl, firstName, lastName, username, speciality,
}) => (
  <Link to={`/trainers/${id}`}>
    <div className="profile-card p-0 m-1">
      <img src={imgUrl} className="img img-responsive h-100 w-100" alt="Trainer" />
      <h3 className="profile-name">{`${firstName} ${lastName}`}</h3>
      <h5 className="profile-username">{username}</h5>
      <h6 className="profile-speciality">{speciality}</h6>
    </div>
  </Link>
);

Trainer.propTypes = {
  id: PropTypes.string.isRequired,
  imgUrl: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  speciality: PropTypes.string.isRequired,
};

export default Trainer;
