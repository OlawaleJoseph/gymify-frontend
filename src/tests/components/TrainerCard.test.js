import Trainer from '../../components/TrainerCard';
import { mountRender } from '../setup';

describe('Side Bar', () => {
  const props = {
    id: 'id',
    imgUrl: 'imageurl',
    firstName: 'John',
    lastName: 'Doe',
    username: 'johndoe',
    speciality: 'speciality',
  };
  const page = mountRender(Trainer, props);

  test('should render trainer', () => {
    expect(page).toBeTruthy();
    expect(page).toMatchSnapshot();
  });

  test('should render a link with trainer id as params', () => {
    const link = page.find('a');

    expect(link).toBeTruthy();
    expect(link.props().href).toEqual(`/trainers/${props.id}`);
  });

  test('should render profile card', () => {
    const card = page.find('.profile-card');

    expect(card).toBeTruthy();
  });

  test('should display trainer image', () => {
    const image = page.find('img');

    expect(image).toBeTruthy();
    expect(image.length).toBe(1);
    expect(image.props().src).toEqual(props.imgUrl);
  });

  test('should display trainer first name and last name', () => {
    const name = page.find('.profile-name');

    expect(name).toBeTruthy();
    expect(name.length).toBe(1);
    expect(name.text()).toEqual(`${props.firstName} ${props.lastName}`);
  });

  test('should display trainer username', () => {
    const username = page.find('.profile-username');

    expect(username).toBeTruthy();
    expect(username.length).toBe(1);
    expect(username.text()).toEqual(props.username);
  });

  test('should display trainer speciality', () => {
    const speciality = page.find('.profile-speciality');

    expect(speciality).toBeTruthy();
    expect(speciality.length).toBe(1);
    expect(speciality.text()).toEqual(props.speciality);
  });
});
