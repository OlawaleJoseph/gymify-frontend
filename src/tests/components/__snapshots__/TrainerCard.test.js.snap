// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side Bar should render trainer 1`] = `ReactWrapper {}`;
